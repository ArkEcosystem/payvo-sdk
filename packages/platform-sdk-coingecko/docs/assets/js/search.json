{"kinds":{"1":"Module","32":"Variable","128":"Class","512":"Constructor","1024":"Property","2048":"Method","2097152":"Object literal"},"rows":[{"id":0,"kind":1,"name":"\"transformers/historical-price-transformer\"","url":"modules/_transformers_historical_price_transformer_.html","classes":"tsd-kind-module"},{"id":1,"kind":128,"name":"HistoricalPriceTransformer","url":"classes/_transformers_historical_price_transformer_.historicalpricetransformer.html","classes":"tsd-kind-class tsd-parent-kind-module","parent":"\"transformers/historical-price-transformer\""},{"id":2,"kind":512,"name":"constructor","url":"classes/_transformers_historical_price_transformer_.historicalpricetransformer.html#constructor","classes":"tsd-kind-constructor tsd-parent-kind-class","parent":"\"transformers/historical-price-transformer\".HistoricalPriceTransformer"},{"id":3,"kind":1024,"name":"data","url":"classes/_transformers_historical_price_transformer_.historicalpricetransformer.html#data","classes":"tsd-kind-property tsd-parent-kind-class tsd-is-private","parent":"\"transformers/historical-price-transformer\".HistoricalPriceTransformer"},{"id":4,"kind":2048,"name":"transform","url":"classes/_transformers_historical_price_transformer_.historicalpricetransformer.html#transform","classes":"tsd-kind-method tsd-parent-kind-class","parent":"\"transformers/historical-price-transformer\".HistoricalPriceTransformer"},{"id":5,"kind":1,"name":"\"transformers/historical-volume-transformer\"","url":"modules/_transformers_historical_volume_transformer_.html","classes":"tsd-kind-module"},{"id":6,"kind":128,"name":"HistoricalVolumeTransformer","url":"classes/_transformers_historical_volume_transformer_.historicalvolumetransformer.html","classes":"tsd-kind-class tsd-parent-kind-module","parent":"\"transformers/historical-volume-transformer\""},{"id":7,"kind":512,"name":"constructor","url":"classes/_transformers_historical_volume_transformer_.historicalvolumetransformer.html#constructor","classes":"tsd-kind-constructor tsd-parent-kind-class","parent":"\"transformers/historical-volume-transformer\".HistoricalVolumeTransformer"},{"id":8,"kind":1024,"name":"data","url":"classes/_transformers_historical_volume_transformer_.historicalvolumetransformer.html#data","classes":"tsd-kind-property tsd-parent-kind-class tsd-is-private","parent":"\"transformers/historical-volume-transformer\".HistoricalVolumeTransformer"},{"id":9,"kind":2048,"name":"transform","url":"classes/_transformers_historical_volume_transformer_.historicalvolumetransformer.html#transform","classes":"tsd-kind-method tsd-parent-kind-class","parent":"\"transformers/historical-volume-transformer\".HistoricalVolumeTransformer"},{"id":10,"kind":1,"name":"\"transformers/market-transformer\"","url":"modules/_transformers_market_transformer_.html","classes":"tsd-kind-module"},{"id":11,"kind":128,"name":"MarketTransformer","url":"classes/_transformers_market_transformer_.markettransformer.html","classes":"tsd-kind-class tsd-parent-kind-module","parent":"\"transformers/market-transformer\""},{"id":12,"kind":512,"name":"constructor","url":"classes/_transformers_market_transformer_.markettransformer.html#constructor","classes":"tsd-kind-constructor tsd-parent-kind-class","parent":"\"transformers/market-transformer\".MarketTransformer"},{"id":13,"kind":1024,"name":"data","url":"classes/_transformers_market_transformer_.markettransformer.html#data","classes":"tsd-kind-property tsd-parent-kind-class tsd-is-private","parent":"\"transformers/market-transformer\".MarketTransformer"},{"id":14,"kind":2048,"name":"transform","url":"classes/_transformers_market_transformer_.markettransformer.html#transform","classes":"tsd-kind-method tsd-parent-kind-class","parent":"\"transformers/market-transformer\".MarketTransformer"},{"id":15,"kind":1,"name":"\"index\"","url":"modules/_index_.html","classes":"tsd-kind-module"},{"id":16,"kind":128,"name":"PriceTracker","url":"classes/_index_.pricetracker.html","classes":"tsd-kind-class tsd-parent-kind-module","parent":"\"index\""},{"id":17,"kind":1024,"name":"tokenLookup","url":"classes/_index_.pricetracker.html#tokenlookup","classes":"tsd-kind-property tsd-parent-kind-class tsd-is-private","parent":"\"index\".PriceTracker"},{"id":18,"kind":1024,"name":"#httpClient","url":"classes/_index_.pricetracker.html#_httpclient","classes":"tsd-kind-property tsd-parent-kind-class tsd-is-private","parent":"\"index\".PriceTracker"},{"id":19,"kind":1024,"name":"#host","url":"classes/_index_.pricetracker.html#_host","classes":"tsd-kind-property tsd-parent-kind-class tsd-is-private","parent":"\"index\".PriceTracker"},{"id":20,"kind":512,"name":"constructor","url":"classes/_index_.pricetracker.html#constructor","classes":"tsd-kind-constructor tsd-parent-kind-class","parent":"\"index\".PriceTracker"},{"id":21,"kind":2048,"name":"verifyToken","url":"classes/_index_.pricetracker.html#verifytoken","classes":"tsd-kind-method tsd-parent-kind-class","parent":"\"index\".PriceTracker"},{"id":22,"kind":2048,"name":"marketData","url":"classes/_index_.pricetracker.html#marketdata","classes":"tsd-kind-method tsd-parent-kind-class","parent":"\"index\".PriceTracker"},{"id":23,"kind":2048,"name":"historicalPrice","url":"classes/_index_.pricetracker.html#historicalprice","classes":"tsd-kind-method tsd-parent-kind-class","parent":"\"index\".PriceTracker"},{"id":24,"kind":2048,"name":"historicalVolume","url":"classes/_index_.pricetracker.html#historicalvolume","classes":"tsd-kind-method tsd-parent-kind-class","parent":"\"index\".PriceTracker"},{"id":25,"kind":2048,"name":"dailyAverage","url":"classes/_index_.pricetracker.html#dailyaverage","classes":"tsd-kind-method tsd-parent-kind-class","parent":"\"index\".PriceTracker"},{"id":26,"kind":2048,"name":"getTokenId","url":"classes/_index_.pricetracker.html#gettokenid","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-private","parent":"\"index\".PriceTracker"},{"id":27,"kind":2048,"name":"get","url":"classes/_index_.pricetracker.html#get","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-private","parent":"\"index\".PriceTracker"},{"id":28,"kind":1,"name":"\"index.test\"","url":"modules/_index_test_.html","classes":"tsd-kind-module"},{"id":29,"kind":32,"name":"BASE_URL_COINGECKO","url":"modules/_index_test_.html#base_url_coingecko","classes":"tsd-kind-variable tsd-parent-kind-module tsd-is-not-exported","parent":"\"index.test\""},{"id":30,"kind":32,"name":"token","url":"modules/_index_test_.html#token","classes":"tsd-kind-variable tsd-parent-kind-module tsd-is-not-exported","parent":"\"index.test\""},{"id":31,"kind":32,"name":"currency","url":"modules/_index_test_.html#currency","classes":"tsd-kind-variable tsd-parent-kind-module tsd-is-not-exported","parent":"\"index.test\""},{"id":32,"kind":32,"name":"subject","url":"modules/_index_test_.html#subject","classes":"tsd-kind-variable tsd-parent-kind-module tsd-is-not-exported","parent":"\"index.test\""},{"id":33,"kind":1,"name":"\"transformers/historical-price-transformer.test\"","url":"modules/_transformers_historical_price_transformer_test_.html","classes":"tsd-kind-module"},{"id":34,"kind":32,"name":"stubResponse","url":"modules/_transformers_historical_price_transformer_test_.html#stubresponse","classes":"tsd-kind-variable tsd-parent-kind-module tsd-is-not-exported","parent":"\"transformers/historical-price-transformer.test\""},{"id":35,"kind":2097152,"name":"stubOptions","url":"modules/_transformers_historical_price_transformer_test_.html#stuboptions","classes":"tsd-kind-object-literal tsd-parent-kind-module tsd-is-not-exported","parent":"\"transformers/historical-price-transformer.test\""},{"id":36,"kind":32,"name":"type","url":"modules/_transformers_historical_price_transformer_test_.html#stuboptions.type","classes":"tsd-kind-variable tsd-parent-kind-object-literal tsd-is-not-exported","parent":"\"transformers/historical-price-transformer.test\".stubOptions"},{"id":37,"kind":32,"name":"dateFormat","url":"modules/_transformers_historical_price_transformer_test_.html#stuboptions.dateformat","classes":"tsd-kind-variable tsd-parent-kind-object-literal tsd-is-not-exported","parent":"\"transformers/historical-price-transformer.test\".stubOptions"},{"id":38,"kind":1,"name":"\"transformers/historical-volume-transformer.test\"","url":"modules/_transformers_historical_volume_transformer_test_.html","classes":"tsd-kind-module"},{"id":39,"kind":32,"name":"stubResponse","url":"modules/_transformers_historical_volume_transformer_test_.html#stubresponse","classes":"tsd-kind-variable tsd-parent-kind-module tsd-is-not-exported","parent":"\"transformers/historical-volume-transformer.test\""},{"id":40,"kind":2097152,"name":"stubOptions","url":"modules/_transformers_historical_volume_transformer_test_.html#stuboptions","classes":"tsd-kind-object-literal tsd-parent-kind-module tsd-is-not-exported","parent":"\"transformers/historical-volume-transformer.test\""},{"id":41,"kind":32,"name":"type","url":"modules/_transformers_historical_volume_transformer_test_.html#stuboptions.type","classes":"tsd-kind-variable tsd-parent-kind-object-literal tsd-is-not-exported","parent":"\"transformers/historical-volume-transformer.test\".stubOptions"},{"id":42,"kind":32,"name":"dateFormat","url":"modules/_transformers_historical_volume_transformer_test_.html#stuboptions.dateformat","classes":"tsd-kind-variable tsd-parent-kind-object-literal tsd-is-not-exported","parent":"\"transformers/historical-volume-transformer.test\".stubOptions"},{"id":43,"kind":1,"name":"\"transformers/market-transformer.test\"","url":"modules/_transformers_market_transformer_test_.html","classes":"tsd-kind-module"},{"id":44,"kind":32,"name":"stubResponse","url":"modules/_transformers_market_transformer_test_.html#stubresponse","classes":"tsd-kind-variable tsd-parent-kind-module tsd-is-not-exported","parent":"\"transformers/market-transformer.test\""},{"id":45,"kind":2097152,"name":"stubOptions","url":"modules/_transformers_market_transformer_test_.html#stuboptions","classes":"tsd-kind-object-literal tsd-parent-kind-module tsd-is-not-exported","parent":"\"transformers/market-transformer.test\""},{"id":46,"kind":32,"name":"type","url":"modules/_transformers_market_transformer_test_.html#stuboptions.type","classes":"tsd-kind-variable tsd-parent-kind-object-literal tsd-is-not-exported","parent":"\"transformers/market-transformer.test\".stubOptions"},{"id":47,"kind":32,"name":"dateFormat","url":"modules/_transformers_market_transformer_test_.html#stuboptions.dateformat","classes":"tsd-kind-variable tsd-parent-kind-object-literal tsd-is-not-exported","parent":"\"transformers/market-transformer.test\".stubOptions"}],"index":{"version":"2.3.9","fields":["name","parent"],"fieldVectors":[["name/0",[0,5.424,1,9.588,2,12.573]],["parent/0",[]],["name/1",[3,37.509]],["parent/1",[0,0.657,1,1.161,2,1.522]],["name/2",[4,25.689]],["parent/2",[0,0.657,1,1.161,5,1.988]],["name/3",[6,28.393]],["parent/3",[0,0.657,1,1.161,5,1.988]],["name/4",[7,28.393]],["parent/4",[0,0.657,1,1.161,5,1.988]],["name/5",[0,5.424,2,12.573,8,9.588]],["parent/5",[]],["name/6",[9,37.509]],["parent/6",[0,0.657,2,1.522,8,1.161]],["name/7",[4,25.689]],["parent/7",[0,0.657,8,1.161,10,1.988]],["name/8",[6,28.393]],["parent/8",[0,0.657,8,1.161,10,1.988]],["name/9",[7,28.393]],["parent/9",[0,0.657,8,1.161,10,1.988]],["name/10",[2,15.93,11,12.148]],["parent/10",[]],["name/11",[12,37.509]],["parent/11",[2,1.867,11,1.424]],["name/12",[4,25.689]],["parent/12",[11,1.424,13,2.439]],["name/13",[6,28.393]],["parent/13",[11,1.424,13,2.439]],["name/14",[7,28.393]],["parent/14",[11,1.424,13,2.439]],["name/15",[14,32.013]],["parent/15",[]],["name/16",[15,37.509]],["parent/16",[14,3.558]],["name/17",[16,37.509]],["parent/17",[17,1.733]],["name/18",[18,37.509]],["parent/18",[17,1.733]],["name/19",[19,37.509]],["parent/19",[17,1.733]],["name/20",[4,25.689]],["parent/20",[17,1.733]],["name/21",[20,37.509]],["parent/21",[17,1.733]],["name/22",[21,37.509]],["parent/22",[17,1.733]],["name/23",[22,37.509]],["parent/23",[17,1.733]],["name/24",[23,37.509]],["parent/24",[17,1.733]],["name/25",[24,37.509]],["parent/25",[17,1.733]],["name/26",[25,37.509]],["parent/26",[17,1.733]],["name/27",[26,37.509]],["parent/27",[17,1.733]],["name/28",[27,23.53]],["parent/28",[]],["name/29",[28,37.509]],["parent/29",[27,2.615]],["name/30",[29,37.509]],["parent/30",[27,2.615]],["name/31",[30,37.509]],["parent/31",[27,2.615]],["name/32",[31,37.509]],["parent/32",[27,2.615]],["name/33",[0,5.424,1,9.588,32,10.211]],["parent/33",[]],["name/34",[33,28.393]],["parent/34",[0,0.657,1,1.161,32,1.236]],["name/35",[34,28.393]],["parent/35",[0,0.657,1,1.161,32,1.236]],["name/36",[35,28.393]],["parent/36",[0,0.657,1,1.161,36,1.522]],["name/37",[37,28.393]],["parent/37",[0,0.657,1,1.161,36,1.522]],["name/38",[0,5.424,8,9.588,32,10.211]],["parent/38",[]],["name/39",[33,28.393]],["parent/39",[0,0.657,8,1.161,32,1.236]],["name/40",[34,28.393]],["parent/40",[0,0.657,8,1.161,32,1.236]],["name/41",[35,28.393]],["parent/41",[0,0.657,8,1.161,36,1.522]],["name/42",[37,28.393]],["parent/42",[0,0.657,8,1.161,36,1.522]],["name/43",[11,12.148,32,12.938]],["parent/43",[]],["name/44",[33,28.393]],["parent/44",[11,1.424,32,1.516]],["name/45",[34,28.393]],["parent/45",[11,1.424,32,1.516]],["name/46",[35,28.393]],["parent/46",[11,1.424,36,1.867]],["name/47",[37,28.393]],["parent/47",[11,1.424,36,1.867]]],"invertedIndex":[["base_url_coingecko",{"_index":28,"name":{"29":{}},"parent":{}}],["constructor",{"_index":4,"name":{"2":{},"7":{},"12":{},"20":{}},"parent":{}}],["currency",{"_index":30,"name":{"31":{}},"parent":{}}],["dailyaverage",{"_index":24,"name":{"25":{}},"parent":{}}],["data",{"_index":6,"name":{"3":{},"8":{},"13":{}},"parent":{}}],["dateformat",{"_index":37,"name":{"37":{},"42":{},"47":{}},"parent":{}}],["get",{"_index":26,"name":{"27":{}},"parent":{}}],["gettokenid",{"_index":25,"name":{"26":{}},"parent":{}}],["historicalprice",{"_index":22,"name":{"23":{}},"parent":{}}],["historicalpricetransformer",{"_index":3,"name":{"1":{}},"parent":{}}],["historicalvolume",{"_index":23,"name":{"24":{}},"parent":{}}],["historicalvolumetransformer",{"_index":9,"name":{"6":{}},"parent":{}}],["host",{"_index":19,"name":{"19":{}},"parent":{}}],["httpclient",{"_index":18,"name":{"18":{}},"parent":{}}],["index",{"_index":14,"name":{"15":{}},"parent":{"16":{}}}],["index\".pricetracker",{"_index":17,"name":{},"parent":{"17":{},"18":{},"19":{},"20":{},"21":{},"22":{},"23":{},"24":{},"25":{},"26":{},"27":{}}}],["index.test",{"_index":27,"name":{"28":{}},"parent":{"29":{},"30":{},"31":{},"32":{}}}],["marketdata",{"_index":21,"name":{"22":{}},"parent":{}}],["markettransformer",{"_index":12,"name":{"11":{}},"parent":{}}],["price",{"_index":1,"name":{"0":{},"33":{}},"parent":{"1":{},"2":{},"3":{},"4":{},"34":{},"35":{},"36":{},"37":{}}}],["pricetracker",{"_index":15,"name":{"16":{}},"parent":{}}],["stuboptions",{"_index":34,"name":{"35":{},"40":{},"45":{}},"parent":{}}],["stubresponse",{"_index":33,"name":{"34":{},"39":{},"44":{}},"parent":{}}],["subject",{"_index":31,"name":{"32":{}},"parent":{}}],["token",{"_index":29,"name":{"30":{}},"parent":{}}],["tokenlookup",{"_index":16,"name":{"17":{}},"parent":{}}],["transform",{"_index":7,"name":{"4":{},"9":{},"14":{}},"parent":{}}],["transformer",{"_index":2,"name":{"0":{},"5":{},"10":{}},"parent":{"1":{},"6":{},"11":{}}}],["transformer\".historicalpricetransformer",{"_index":5,"name":{},"parent":{"2":{},"3":{},"4":{}}}],["transformer\".historicalvolumetransformer",{"_index":10,"name":{},"parent":{"7":{},"8":{},"9":{}}}],["transformer\".markettransformer",{"_index":13,"name":{},"parent":{"12":{},"13":{},"14":{}}}],["transformer.test",{"_index":32,"name":{"33":{},"38":{},"43":{}},"parent":{"34":{},"35":{},"39":{},"40":{},"44":{},"45":{}}}],["transformer.test\".stuboptions",{"_index":36,"name":{},"parent":{"36":{},"37":{},"41":{},"42":{},"46":{},"47":{}}}],["transformers/historical",{"_index":0,"name":{"0":{},"5":{},"33":{},"38":{}},"parent":{"1":{},"2":{},"3":{},"4":{},"6":{},"7":{},"8":{},"9":{},"34":{},"35":{},"36":{},"37":{},"39":{},"40":{},"41":{},"42":{}}}],["transformers/market",{"_index":11,"name":{"10":{},"43":{}},"parent":{"11":{},"12":{},"13":{},"14":{},"44":{},"45":{},"46":{},"47":{}}}],["type",{"_index":35,"name":{"36":{},"41":{},"46":{}},"parent":{}}],["verifytoken",{"_index":20,"name":{"21":{}},"parent":{}}],["volume",{"_index":8,"name":{"5":{},"38":{}},"parent":{"6":{},"7":{},"8":{},"9":{},"39":{},"40":{},"41":{},"42":{}}}]],"pipeline":[]}}