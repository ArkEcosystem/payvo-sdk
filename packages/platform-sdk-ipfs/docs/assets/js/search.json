{"kinds":{"1":"Module","128":"Class","256":"Interface","512":"Constructor","1024":"Property","2048":"Method","4194304":"Type alias","16777216":"Reference"},"rows":[{"id":0,"kind":1,"name":"\"file.models\"","url":"modules/_file_models_.html","classes":"tsd-kind-module"},{"id":1,"kind":256,"name":"FileData","url":"interfaces/_file_models_.filedata.html","classes":"tsd-kind-interface tsd-parent-kind-module","parent":"\"file.models\""},{"id":2,"kind":1024,"name":"path","url":"interfaces/_file_models_.filedata.html#path","classes":"tsd-kind-property tsd-parent-kind-interface","parent":"\"file.models\".FileData"},{"id":3,"kind":1024,"name":"content","url":"interfaces/_file_models_.filedata.html#content","classes":"tsd-kind-property tsd-parent-kind-interface","parent":"\"file.models\".FileData"},{"id":4,"kind":4194304,"name":"FileMap","url":"modules/_file_models_.html#filemap","classes":"tsd-kind-type-alias tsd-parent-kind-module","parent":"\"file.models\""},{"id":5,"kind":1,"name":"\"file\"","url":"modules/_file_.html","classes":"tsd-kind-module"},{"id":6,"kind":128,"name":"File","url":"classes/_file_.file.html","classes":"tsd-kind-class tsd-parent-kind-module","parent":"\"file\""},{"id":7,"kind":1024,"name":"#client","url":"classes/_file_.file.html#_client","classes":"tsd-kind-property tsd-parent-kind-class tsd-is-private","parent":"\"file\".File"},{"id":8,"kind":512,"name":"constructor","url":"classes/_file_.file.html#constructor","classes":"tsd-kind-constructor tsd-parent-kind-class","parent":"\"file\".File"},{"id":9,"kind":2048,"name":"make","url":"classes/_file_.file.html#make","classes":"tsd-kind-method tsd-parent-kind-class","parent":"\"file\".File"},{"id":10,"kind":2048,"name":"upload","url":"classes/_file_.file.html#upload","classes":"tsd-kind-method tsd-parent-kind-class","parent":"\"file\".File"},{"id":11,"kind":1,"name":"\"link\"","url":"modules/_link_.html","classes":"tsd-kind-module"},{"id":12,"kind":128,"name":"Link","url":"classes/_link_.link.html","classes":"tsd-kind-class tsd-parent-kind-module","parent":"\"link\""},{"id":13,"kind":2048,"name":"ipfs","url":"classes/_link_.link.html#ipfs","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-static","parent":"\"link\".Link"},{"id":14,"kind":2048,"name":"cloudflare","url":"classes/_link_.link.html#cloudflare","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-static","parent":"\"link\".Link"},{"id":15,"kind":1,"name":"\"index\"","url":"modules/_index_.html","classes":"tsd-kind-module"},{"id":16,"kind":16777216,"name":"File","url":"modules/_index_.html#file","classes":"tsd-kind-reference tsd-parent-kind-module","parent":"\"index\""},{"id":17,"kind":16777216,"name":"Link","url":"modules/_index_.html#link","classes":"tsd-kind-reference tsd-parent-kind-module","parent":"\"index\""}],"index":{"version":"2.3.9","fields":["name","parent"],"fieldVectors":[["name/0",[0,16.917]],["parent/0",[]],["name/1",[1,25.39]],["parent/1",[0,1.515]],["name/2",[2,25.39]],["parent/2",[3,1.816]],["name/3",[4,25.39]],["parent/3",[3,1.816]],["name/4",[5,25.39]],["parent/4",[0,1.515]],["name/5",[6,14.404]],["parent/5",[]],["name/6",[6,14.404]],["parent/6",[6,1.29]],["name/7",[7,25.39]],["parent/7",[8,1.29]],["name/8",[9,25.39]],["parent/8",[8,1.29]],["name/9",[10,25.39]],["parent/9",[8,1.29]],["name/10",[11,25.39]],["parent/10",[8,1.29]],["name/11",[12,14.404]],["parent/11",[]],["name/12",[12,14.404]],["parent/12",[12,1.29]],["name/13",[13,25.39]],["parent/13",[14,1.816]],["name/14",[15,25.39]],["parent/14",[14,1.816]],["name/15",[16,16.917]],["parent/15",[]],["name/16",[6,14.404]],["parent/16",[16,1.515]],["name/17",[12,14.404]],["parent/17",[16,1.515]]],"invertedIndex":[["client",{"_index":7,"name":{"7":{}},"parent":{}}],["cloudflare",{"_index":15,"name":{"14":{}},"parent":{}}],["constructor",{"_index":9,"name":{"8":{}},"parent":{}}],["content",{"_index":4,"name":{"3":{}},"parent":{}}],["file",{"_index":6,"name":{"5":{},"6":{},"16":{}},"parent":{"6":{}}}],["file\".file",{"_index":8,"name":{},"parent":{"7":{},"8":{},"9":{},"10":{}}}],["file.models",{"_index":0,"name":{"0":{}},"parent":{"1":{},"4":{}}}],["file.models\".filedata",{"_index":3,"name":{},"parent":{"2":{},"3":{}}}],["filedata",{"_index":1,"name":{"1":{}},"parent":{}}],["filemap",{"_index":5,"name":{"4":{}},"parent":{}}],["index",{"_index":16,"name":{"15":{}},"parent":{"16":{},"17":{}}}],["ipfs",{"_index":13,"name":{"13":{}},"parent":{}}],["link",{"_index":12,"name":{"11":{},"12":{},"17":{}},"parent":{"12":{}}}],["link\".link",{"_index":14,"name":{},"parent":{"13":{},"14":{}}}],["make",{"_index":10,"name":{"9":{}},"parent":{}}],["path",{"_index":2,"name":{"2":{}},"parent":{}}],["upload",{"_index":11,"name":{"10":{}},"parent":{}}]],"pipeline":[]}}