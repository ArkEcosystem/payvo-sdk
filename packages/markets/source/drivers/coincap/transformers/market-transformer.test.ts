import { MarketTransformer } from "./market-transformer.js";

const stubOptions = { type: "day", dateFormat: "DD.MM", token: "ARK" };

describe("MarketTransformer", () => {
	it("should transform the given data", async () => {
		const stubResponse = (await import("../../../../test/fixtures/coincap/market.json")).default;

		const subject = new MarketTransformer(stubResponse);

		expect(subject.transform(stubOptions)).toMatchSnapshot();
	});

	it("should skip unknown currencies", async () => {
		const stubResponse = (await import("../../../../test/fixtures/coincap/market.json")).default;

		const subject = new MarketTransformer(stubResponse);

		expect(subject.transform({ ...stubOptions, currencies: { invalid: {} } })).toMatchSnapshot();
	});
});
