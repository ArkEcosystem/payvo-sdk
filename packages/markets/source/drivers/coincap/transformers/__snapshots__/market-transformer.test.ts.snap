// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketTransformer should skip unknown currencies 1`] = `Object {}`;

exports[`MarketTransformer should transform the given data 1`] = `
Object {
  "AUD": Object {
    "change24h": null,
    "currency": "AUD",
    "date": 2020-02-20T06:08:12.727Z,
    "marketCap": 40049076.952300504,
    "price": 0.3386501157195427,
    "volume": 801594.3509484902,
  },
  "BRL": Object {
    "change24h": null,
    "currency": "BRL",
    "date": 2020-02-20T06:08:12.727Z,
    "marketCap": 116292755.73684944,
    "price": 0.9833573751158916,
    "volume": 2327634.5711018043,
  },
  "BTC": Object {
    "change24h": null,
    "currency": "BTC",
    "date": 2020-02-20T06:08:12.727Z,
    "marketCap": 2781.8373818836935,
    "price": 0.000023522878003153278,
    "volume": 55.679313988464244,
  },
  "CAD": Object {
    "change24h": null,
    "currency": "CAD",
    "date": 2020-02-20T06:08:12.727Z,
    "marketCap": 35243716.224592574,
    "price": 0.2980165708203522,
    "volume": 705413.6070529897,
  },
  "CHF": Object {
    "change24h": null,
    "currency": "CHF",
    "date": 2020-02-20T06:08:12.727Z,
    "marketCap": 26202657.72076011,
    "price": 0.22156648154122477,
    "volume": 524454.0950048447,
  },
  "CNY": Object {
    "change24h": null,
    "currency": "CNY",
    "date": 2020-02-20T06:08:12.727Z,
    "marketCap": 186945879.2572925,
    "price": 1.5807915802697956,
    "volume": 3741778.1419588053,
  },
  "DKK": Object {
    "change24h": null,
    "currency": "DKK",
    "date": 2020-02-20T06:08:12.727Z,
    "marketCap": 184279132.07842582,
    "price": 1.5582418910024605,
    "volume": 3688402.3930861573,
  },
  "ETH": Object {
    "change24h": null,
    "currency": "ETH",
    "date": 2020-02-20T06:08:12.727Z,
    "marketCap": 102505.23035786345,
    "price": 0.0008667717401800148,
    "volume": 2051.673093374931,
  },
  "EUR": Object {
    "change24h": null,
    "currency": "EUR",
    "date": 2020-02-20T06:08:12.727Z,
    "marketCap": 24672652.5167786,
    "price": 0.208628943929635,
    "volume": 493830.5795142356,
  },
  "GBP": Object {
    "change24h": null,
    "currency": "GBP",
    "date": 2020-02-20T06:08:12.727Z,
    "marketCap": 20633200.097881805,
    "price": 0.1744718263827746,
    "volume": 412979.72135914175,
  },
  "HKD": Object {
    "change24h": null,
    "currency": "HKD",
    "date": 2020-02-20T06:08:12.727Z,
    "marketCap": 207105714.34934595,
    "price": 1.7512606898310996,
    "volume": 4145283.3199954974,
  },
  "IDR": Object {
    "change24h": null,
    "currency": "IDR",
    "date": 2020-02-20T06:08:12.727Z,
    "marketCap": 364442923708.87976,
    "price": 3081.6849645292805,
    "volume": 7294435006.233239,
  },
  "INR": Object {
    "change24h": null,
    "currency": "INR",
    "date": 2020-02-20T06:08:12.727Z,
    "marketCap": 1909038313.8157303,
    "price": 16.142595412541752,
    "volume": 38209977.4714288,
  },
  "JPY": Object {
    "change24h": null,
    "currency": "JPY",
    "date": 2020-02-20T06:08:12.727Z,
    "marketCap": 2967409496.446644,
    "price": 25.092053196526955,
    "volume": 59393595.8158431,
  },
  "KRW": Object {
    "change24h": null,
    "currency": "KRW",
    "date": 2020-02-20T06:08:12.727Z,
    "marketCap": 31937886330.204124,
    "price": 270.0628759669815,
    "volume": 639246424.9305512,
  },
  "LTC": Object {
    "change24h": null,
    "currency": "LTC",
    "date": 2020-02-20T06:08:12.727Z,
    "marketCap": 377925.91656455386,
    "price": 0.0031956955095477847,
    "volume": 7564.301173682234,
  },
  "MXN": Object {
    "change24h": null,
    "currency": "MXN",
    "date": 2020-02-20T06:08:12.727Z,
    "marketCap": 495868090.93368065,
    "price": 4.193000167677364,
    "volume": 9924949.356047973,
  },
  "NOK": Object {
    "change24h": null,
    "currency": "NOK",
    "date": 2020-02-20T06:08:12.727Z,
    "marketCap": 247653542.7368812,
    "price": 2.094128186926465,
    "volume": 4956860.3313056175,
  },
  "RUB": Object {
    "change24h": null,
    "currency": "RUB",
    "date": 2020-02-20T06:08:12.727Z,
    "marketCap": 1696175660.5316842,
    "price": 14.342654748419395,
    "volume": 33949467.28280141,
  },
  "SEK": Object {
    "change24h": null,
    "currency": "SEK",
    "date": 2020-02-20T06:08:12.727Z,
    "marketCap": 261089383.75599456,
    "price": 2.2077400217592964,
    "volume": 5225782.74052845,
  },
  "USD": Object {
    "change24h": null,
    "currency": "USD",
    "date": 2020-02-20T06:08:12.727Z,
    "marketCap": 26638435.893542577,
    "price": 0.2252513686814852,
    "volume": 533176.3265305582,
  },
  "VND": Object {
    "change24h": null,
    "currency": "VND",
    "date": 2020-02-20T06:08:12.727Z,
    "marketCap": 618497691541.3165,
    "price": 5229.941131032868,
    "volume": 12379417787.947847,
  },
}
`;
